app/services/imo_api_service.rb

```diff
@@ -52,11 +53,13 @@ def send_notification(patient, message)
     response = execute_post(url, body: request_body)
     result = process_response(response, url, "notification")
 
-    return if result.nil?
-
-    unless patient.imo_authorization.status == result.to_s
-      patient.imo_authorization.update!(status: result)
+    # until we implement the invitation callback, the only way for us to know if the user
+    # has accepted our invitation is to send a notication to see if it succeeds
+    status = result == :success ? :subscribed : result
+    unless patient.imo_authorization.status == status.to_s || status == :error
+      patient.imo_authorization.update!(status: status)
```

Comment:
It might be a good idea to extract this chunk into a semantic method, especially since this is a temporary "btw has this user accepted our invite?" hook. Perhaps something like `update_imo_subscription_status(result)`?

Reply:
Sorry, this is one of those things that's just captured in my head. When I started work on this, there was no invitation callback, so the only way to know if the user was subscribed was to send a message and see if it succeeds. This changed last week when Imo added an invitation callback. So this logic will change in the near future. We will only send notifications to patients who have subscribed, so this logic will be removed. I left this here because I haven't been able to test the invitation callback yet to confirm that it works. I was not able to test that due to technical constraints on Imo's side. They've just unblocked me (I believe), so I will test it soon. Regardless, I suggest we leave this here for now and I'll change it while I'm adding the invitation callback logic.